site/src/sphinx/advanced-retry.rst

```diff
@@ -1,8 +1,154 @@
+.. _RetryingClient: apidocs/index.html?com/linecorp/armeria/client/retry/RetryingClient.html
+.. _RetryingHttpClient: apidocs/index.html?com/linecorp/armeria/client/retry/RetryingHttpClient.html
+.. _RetryingRpcClient: apidocs/index.html?com/linecorp/armeria/client/retry/RetryingRpcClient.html
+.. _ClientBuilder: apidocs/index.html?com/linecorp/armeria/client/ClientBuilder.html
+.. _RetryStrategy: apidocs/index.html?com/linecorp/armeria/client/retry/RetryStrategy.html
+.. _Backoff: apidocs/index.html?com/linecorp/armeria/client/retry/Backoff.html
+.. _FixedBackoff: apidocs/index.html?com/linecorp/armeria/client/retry/Backoff.html
+.. _RandomBackoff: apidocs/index.html?com/linecorp/armeria/client/retry/Backoff.html
+.. _ExponentialBackoff: apidocs/index.html?com/linecorp/armeria/client/retry/Backoff.html
 .. _com.linecorp.armeria.client.retry: apidocs/index.html?com/linecorp/armeria/client/retry/package-summary.html
+.. _LoggingClient: apidocs/index.html?com/linecorp/armeria/client/logging/LoggingClient.html
 
 .. _advanced-retry:
 
 Automatic retry
 ===============
 
-TBW - See `com.linecorp.armeria.client.retry`_.
+When a client gets error response, the client might want to retry the request depending on the response.
+You can do this using :ref:`client-decorator` or use RetryingClient_ which already provides the
+retrying functionality in Armeria.
+
+There are two retrying clients:
+
+- RetryingHttpClient_
+- RetryingRpcClient_
+
+They are same except that they have the different request and response types.
+So, let's find out what we can do with RetryingClient_.
+
+RetryingClient_
+---------------
+
+You can just use ``decorator`` method in ClientBuilder_ to build a RetryingHttpClient_:
+
+.. code-block:: java
+
+    RetryStrategy strategy = RetryStrategy.onServerErrorStatus();
```

Comment:
Could you add some `import` statements for first-appearning types in all examples in this document? For example, in this example, you could add `import` for `RetryStratrgy` and `RetryingHttpClient`, and in the next example, you wouldn't need to add any `import`s since they appeared above.